encode.go

```diff
@@ -52,19 +53,42 @@ type tokenData struct {
 func (tokens *tokenData) recursiveEncode(hm interface{}) {
 	v := reflect.ValueOf(hm)
 
+	regexpAttrBox := regexp.MustCompile(`([^\[]+)\[(.*)\]`)
+	regexpAttrs := regexp.MustCompile(`,?([^=]+)=([^,]+)`)
+
 	switch v.Kind() {
 	case reflect.Map:
 		for _, key := range v.MapKeys() {
-			t := xml.StartElement{
-				Name: xml.Name{
-					Space: "",
-					Local: key.String(),
-				},
+			if regexpAttrBox.MatchString(key.String()) {
+				rs := regexpAttrBox.FindStringSubmatch(key.String())
+
+				var attrs []xml.Attr
+				matches := regexpAttrs.FindAllStringSubmatch(rs[2], -1)
+				for _, match := range matches {
+					attrs = append(attrs, xml.Attr{Name: xml.Name{Local: match[1]}, Value: match[2]})
```

Comment:
Since you know the size of ```matches```, please create the variable ```attrs``` using that information.